File: 0605-can-place-flowers/0605-can-place-flowers.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        int nums = flowerbed.Length;
        for(int i = 0; i < nums; i++){
            bool left = i == 0 || flowerbed[i -1] == 0;
            bool right = i == nums-1  ||  flowerbed[i+1] == 0;

            if(right && left && flowerbed[i] == 0){
                flowerbed[i] = 1;
                n--;
            }
        }    
        return n <= 0;     
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
     let count = flowerbed.length;
     for(let i = 0; i < count; i++ ){
        let left = i === 0 || flowerbed[i-1] === 0
        let right = i === count -1 || flowerbed[i+1] === 0;

        if(left && right && flowerbed[i] === 0){
            flowerbed[i] = 1;
            n--;
        } 
     }
     return n <= 0;
};